code/modules/events/crystal_event.dm

```diff
@@ -109,14 +109,28 @@ This section is for the event controller
 	for(var/i in 1 to 6)
 		spawn_anomaly(spawners)
 
-	addtimer(CALLBACK(src, .proc/more_portals, GLOB.crystal_invasion_waves[wave_name]), 15 MINUTES)
+	var/list/crystal_spawner_turfs = list()
+	for(var/turf in RANGE_TURFS(6, dest_crystal.loc))
+		if(!isopenturf(turf) || isspaceturf(turf))
+			continue
+		var/turf/floor = turf
+		crystal_spawner_turfs += floor
+	for(var/i = 0, i < 6, i++)
+		if(!length(crystal_spawner_turfs))
+			break
+		var/pick_portal = pickweight(GLOB.crystal_invasion_waves["big wave"])
+		var/turf/crystal_spawner_turf = pick_n_take(crystal_spawner_turfs)
+		new pick_portal(crystal_spawner_turf)
+
+	addtimer(CALLBACK(src, .proc/more_portals, GLOB.crystal_invasion_waves[wave_name]), 10 MINUTES)
 
 ///Spawn an anomaly randomly in a different location than spawn_portal()
 /datum/round_event/crystal_invasion/proc/spawn_anomaly(list/spawners)
 	if(!spawners.len)
 		CRASH("No landmarks on the station map, aborting")
 	var/obj/spawner = pick(spawners)
-	new/obj/effect/anomaly/flux(spawner.loc)
+	var/obj/effect/anomaly/flux/spawned_anomaly = new(spawner.loc, 300, FALSE)
+	spawned_anomaly.explosive = FALSE
```

Comment:
Could you set this variable during `Initialization()`, passing this option as an argument?